src/hutch_bunny/core/query_solvers.py

```diff
@@ -105,88 +87,303 @@ def _find_concepts(self) -> dict:
+
+                    #firstly add the person constrains
+                    for rule_index, all_constraints_for_person in enumerate(person_constraints_for_group, start=0):
+                        all_parameters.append(all_constraints_for_person)
+
+                    # to get all the constraints in one list, we have to unpack the top-level grouping
+                    # list_for_rules contains all the group of constraints for each rule
+                    # therefore, we get each group, then for each group, we get each constraint
+                    for rule_index, current_rule in enumerate(list_for_rules, start=0):
+                        for constraint_index, current_constraint in enumerate(current_rule, start=0):
+                            all_parameters.append(current_constraint)
+
+                    # all added as an OR
+                    group_query: select = select(Person.person_id).where(or_(*all_parameters))
+
+                #store the query for the given group in the list for assembly later across all groups
+                all_groups_queries.append(Person.person_id.in_(group_query))
+
+            """
+            ALL GROUPS COMPLETED, NOW APPLY LOGIC BETWEEN GROUPS
+            """
+
+            # construct the query based on the OR/AND logic specified between groups
+            if self.query.cohort.groups_operator == "OR":
+                full_query_all_groups = select(func.count()).where(or_(*all_groups_queries))
+            else:
+                full_query_all_groups = select(func.count()).where(*all_groups_queries)
+
+            # here for debug, prints the SQL statement created
+            print(str(full_query_all_groups.compile(
+                dialect=postgresql.dialect(),
+                compile_kwargs={"literal_binds": True})))
+
+            output = con.execute(full_query_all_groups).fetchone()
+
+        return int(output[0])
+
+    def calc_relative_date(self, left_value_time: str, right_value_time: str) -> datetime:
+
+        time_value_supplied: str
+
+        # have to toggle between left and right, given |1 means less than 1 and
+        # 1| means greater than 1
+        if left_value_time == "":
+            time_value_supplied = right_value_time
+        else:
+            time_value_supplied = left_value_time
+        # converting supplied time (in months) (stored as string) to int, and negating.
+        time_to_use: int = int(time_value_supplied)
+        time_to_use = time_to_use * -1
 
+        # the relative date to search on, is the current date minus
+        # the number of months supplied
+        today_date: datetime = datetime.now()
+        relative_date = today_date + relativedelta(months=time_to_use)
+
+        return relative_date
+
+    def add_person_constraints(self, person_constraints_for_group, current_rule:Rule, concepts):
+        concept_domain: str = concepts.get(current_rule.value)
+
+        if concept_domain == "Gender":
```

Comment:
Instead of the nested if..if..else

Could do this with an initial if on the `concept_domain` that selects the correct Person property to a value.

And then an if else on the operator.

Improves the readability of what is happening here